### PR TITLE
add subscription-manager's --type, --consumerid, --name option support

### PR DESCRIPTION
**Pull request by ssato**
This change add support of --type, --name and --consumerid options of subscription-manager necessary for some cases such like registering clients to RHN with exsiting system entries with ---consumerid option.

### CODE REVIEW

**Comment by gregdek** [comment]
@barnabycourt Please review this PR to make sure it adheres to the following guidelines:

http://docs.ansible.com/developing_modules.html#module-checklist

If it passes these guidelines, and if you believe it’s a good PR otherwise, please add a comment with "shipit" in the text, and we will flag it for inclusion.

If it fails these guidelines, please add a comment with "needs_revision" in the text, along with the needed changes, and we will flag it for revision.

**Comment by gregdek** [comment]
@barnabycourt This change is still pending your review; do you have time to take a look and comment? Please comment with text 'shipit' or 'needs_revision' as appropriate.

File: packaging/os/redhat_subscription.py [hunk 1]
```diff
@@ -65,12 +65,34 @@
             - Specify a subscription pool name to consume.  Regular expressions accepted.
         required: False
         default: '^$'
+    consumer_type:
+        description:
+            - The type of unit to register, defaults to system
+        required: False
+        default: null
```
Comment (barnabycourt) [inline packaging/os/redhat_subscription.py]: default value is None and not null

File: packaging/os/redhat_subscription.py [hunk 2]
```diff
@@ -65,12 +65,34 @@
...
+        version_added: "2.x"  # TBD
+    consumer_name:
+        description:
+            - Name of the system to register, defaults to the hostname
+        required: False
+        default: null
```
Comment (barnabycourt) [inline packaging/os/redhat_subscription.py]: default value is None and not null

File: packaging/os/redhat_subscription.py [hunk 3]
```diff
@@ -65,12 +65,34 @@
...
+        version_added: "2.x"  # TBD
+    consumer_id:
+        description:
+            - The existing system data is pulled from the server
+        required: False
+        default: null
```
Comment (barnabycourt) [inline packaging/os/redhat_subscription.py]: default value is None and not null

File: packaging/os/redhat_subscription.py [hunk 4]
```diff
@@ -65,12 +65,34 @@
...
+        version_added: "2.x"  # TBD
+    consumer_name:
+        description:
+            - Name of the system to register, defaults to the hostname
+        required: False
+        default: null
+        version_added: "2.x"  # TBD
+    consumer_id:
+        description:
+            - The existing system data is pulled from the server
```
Comment (barnabycourt) [inline packaging/os/redhat_subscription.py:82]: Should have be more descriptive of what the consumer id is in addition to the effect of specifying it.

**Comment by barnabycourt** [comment]
needs_revision
@ssato Overall looks great, just one small item.

**Comment by robynbergeron** [comment]
Thanks @ssato for this PR. A maintainer of this module has asked for revisions to this PR. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

**Comment by alikins** [comment]
There is a rebased version of this with a more details consumerid docs (addressing https://github.com/ansible/ansible-modules-core/pull/2476#discussion_r48280700) at https://github.com/alikins/ansible-modules-core/tree/alikins_rhsm_options

(Whats the recommended approach for submitting changes to prs? sub-pr on the orig pr? new pr?)

**Comment by ssato** [comment]
@alikins I looked at your version. It surpasses my version and much better! Thanks a lot for the info. Maybe another new PR is needed because your version covers more (not only consumer\* options like my one), I think. I'll close this PR if you want, so please let me know.

**Comment by gregdek** [comment]
@ssato A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes.

**Comment by alikins** [comment]
https://github.com/ansible/ansible-modules-core/pull/3348 is a new pr with the changes from @ssato and a rebase and a few doc tweaks.

**Comment by alikins** [comment]
> @alikins I looked at your version. It surpasses my version and much better! Thanks a lot for the info. Maybe another new PR is needed because your version covers more (not only consumer\* options like my one), I think. I'll close this PR if you want, so please let me know.

I opened another up #3348, so this can be closed now.

**Comment by gregdek** [comment]
@ssato Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by ssato** [comment]
#3348 contain this and closed this.

**Pull request closed without merging**